### PR TITLE
feat!: Remove subscription license signal and data

### PR DESCRIPTION
**Pull request by timmc-edx**
See https://github.com/edx/edx-arch-experiments/issues/41 for related deletions.

**Merge checklist:**
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are squashed

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 1.0.0
+current_version = 2.0.0
```
[the remark this answers is not in the document]
Comment (timmc-edx) [reply setup.cfg:2]: It's something that eduNEXT uses, I think—it's supposed to coordinate bumping the version in multiple places in the repo. I ran it here with `bumpversion --major --no-commit --no-tag` but in the past I think people have just edited the init file directly and not seen this file. I'm not sure if eduNEXT will be keeping it for this repo?

[the remark this answers is not in the document]
Comment (rgraber) [reply setup.cfg:2]: huh, go figure. I'm assuming no harm no foul here (I usually just manually update __init__)

**Review by rgraber — APPROVED** [review]
LGTM

File: setup.cfg (hunk 1 above)
[the remark this answers is not in the document]
Comment (mariajgrimaldi) [reply setup.cfg:2]: We've been using bumpversion to tag our repos and manage versions within the repo (so we don't have to do it manually and don't leave any place out of date). I'll be opening an issue soon so we can set a standard for this, whether removing it or adapting it. Thanks either way for updating the file!

[the remark this answers is not in the document]
Comment (timmc-edx) [reply setup.cfg:2]: OK, good to know!

Also, would you or someone else from eduNEXT be able to review this PR? It's not urgent, just cleanup work from a branch of event-bus work that was later dropped.

[the remark this answers is not in the document]
Comment (mariajgrimaldi) [reply setup.cfg:2]: I'll do it tomorrow! :woman_astronaut:

**Review by mariajgrimaldi — APPROVED** [review]
This looks good :) Thanks for taking care of the cleanup on this side. The issue left to resolve is managing code removal without the codecov/project complaining.
I remember this happening to me once, and I added more tests... which is unsustainable. Do you know how this is handled elsewhere?

**Comment by timmc-edx** [comment]
I think codecov just doesn't handle code-removal situations gracefully -- the amount of tested code has gone down, so the *percentage* of untested code has gone up. I don't think it should fail on that.

In this case, it might be that we want to add a no-coverage comment to this line, which is just in a test helper function: https://github.com/openedx/openedx-events/blob/702411f966d04a235cb26ebf56cd53d0fc82f4cd/openedx_events/event_bus/avro/tests/test_avro.py#L60

**Comment by timmc-edx** [comment]
(In other projects I've just been ignoring these "tiny drop in coverage" failures and merging anyhow.)

**Comment by timmc-edx** [comment]
Merging this for now, and will see if I can increase the test coverage in another PR.

**Pull request merged**